### PR TITLE
[serve] Fix get endpoint when autoscaling config is set

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

If autoscaling config is set for a deployment, we can't set the num replicas when returning the deployment details of that deployment. Otherwise, it breaks the entirety of the get metadata endpoint.

## Related issue number

https://github.com/ray-project/ray/issues/34376

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/schema.py [hunk 1]
```diff
@@ -287,6 +285,13 @@ def _deployment_info_to_schema(name: str, info: DeploymentInfo) -> DeploymentSch
         is_driver_deployment=info.is_driver_deployment,
     )
 
+    if info.deployment_config.autoscaling_config is not None:
```
Comment (tchordia) [inline python/ray/serve/schema.py:288]: if DeploymentSchema is a pydantic model, should we update the pydantic validator? why we are adding this logic here

**Comment by zhe-thoughts** [comment]
@edoakes I'd like your approval here. Thanks

File: python/ray/serve/schema.py (hunk 1 above)
Comment (zcin) [reply python/ray/serve/schema.py:288]: This is a user-facing model for specifying config options for a deployment. The two places the check used currently is:
- when a user submits a config to Serve
- returned here as a copy of the config the user submitted, except with default values all filled in
I think the validator makes sense in these 2 cases.

**Review by sihanwang41 — APPROVED** [review]

File: python/ray/serve/schema.py (hunk 1 above)
Comment (tchordia) [reply python/ray/serve/schema.py:288]: i think standard practice in these cases is to have separate create models. Eg. you can do 
```
class DeploymentConfig(BaseModel):
    ...

class CreateDeploymentConfig(DeploymentConfig):
    # differences for the create model
```
generally speaking it's not good practice to use the same model for create / get paths for this class of reasons.

**Review by edoakes — APPROVED** [review]
stamp + should be release blocker

File: python/ray/serve/schema.py (hunk 1 above)
Comment (zcin) [reply python/ray/serve/schema.py:288]: Got it, that makes sense. I've created a separate issue to track this: https://github.com/ray-project/ray/issues/34383.

**Comment by zhe-thoughts** [comment]
Approved for cherry picking

**Comment by zhe-thoughts** [comment]
@zcin @sihanwang41 @edoakes Quick postmortem question: why wasn't this caught in release testing?

**Pull request merged**